carapace-server/src/test/java/org/carapaceproxy/server/RequestsLoggerTest.java

```diff
@@ -529,4 +523,62 @@ public void testWithServer() throws Exception {
+    public void testAccessLogRotation() throws Exception {
+        stubFor(get(urlEqualTo("/index.html"))
+                .willReturn(aResponse()
+                        .withStatus(200)
+                        .withHeader("Content-Type", "text/html")
+                        .withHeader("Content-Length", "it <b>works</b> !!".length() + "")
+                        .withBody("it <b>works</b> !!")));
+
+        TestEndpointMapper mapper = new TestEndpointMapper("localhost", wireMockRule.port(), true);
+        EndpointKey key = new EndpointKey("localhost", wireMockRule.port());
+
+        RuntimeServerConfiguration c = genConf();
+        c.setAccessLogPath(tmpDir.getRoot().getAbsolutePath() + "/access.log");
+        RequestsLogger reqLogger = new RequestsLogger(c);
+        reqLogger.setVerbose(true);
+
+        try ( HttpProxyServer server = HttpProxyServer.buildForTests("localhost", 0, mapper, tmpDir.newFolder());) {
+            server.getCurrentConfiguration().setAccessLogPath(tmpDir.getRoot().getAbsolutePath() + "/access.log");
+            server.getCurrentConfiguration().setAccessLogMaxSize(1024);
+            Path currentAccessLogPath = Paths.get(server.getCurrentConfiguration().getAccessLogPath());
+            server.start();
+            int port = server.getLocalPort();
+            FileChannel LogFileChannel = FileChannel.open(currentAccessLogPath);
+
+            while (LogFileChannel.size() < 1024) {
+                try ( RawHttpClient client = new RawHttpClient("localhost", port)) {
+                    RawHttpClient.HttpResponse resp = client.executeRequest("GET /index.html HTTP/1.1\r\nHost: localhost\r\nConnection: close\r\n\r\n");
+                    String s = resp.toString();
+                    assertTrue(s.endsWith("it <b>works</b> !!"));
+                }
+
+                try ( RawHttpClient client = new RawHttpClient("localhost", port)) {
+                    {
+                        RawHttpClient.HttpResponse resp = client.executeRequest("GET /index.html HTTP/1.1\r\nHost: localhost\r\n\r\n");
+                        String s = resp.toString();
+                        assertTrue(s.endsWith("it <b>works</b> !!"));
+                    }
+
+                    {
+                        RawHttpClient.HttpResponse resp = client.executeRequest("GET /index.html HTTP/1.1\r\nHost: localhost\r\n\r\n");
+                        String s = resp.toString();
+                        assertTrue(s.endsWith("it <b>works</b> !!"));
+                    }
+                }
+            }
+
+            // test to see if a file exists
+            File file1 = new File(tmpDir.getRoot().getAbsolutePath() + "/access.log");
+            assertTrue(file1.exists());
+            
+            reqLogger.rotateAccessLogFile();
+            
+            File[] f = new File(tmpDir.getRoot().getAbsolutePath()).listFiles((dir, name) -> name.startsWith("access") && name.endsWith(".gzip"));
+            assertEquals(f.length,1);
```

Comment:
Maybe we should check that logs properly works even after a rotation has been performed